modules/tinymce/tools/modules/grunt-webpack.js

```diff
@@ -60,85 +89,101 @@ let create = (entries, tsConfig, outDir, filename) => {
               loader: 'ts-loader',
               options: {
+                configFile: tsConfigPath,
                 transpileOnly: true,
-                compilerOptions: {
-                  declarationMap: false
-                },
-                configFile: tsConfig,
-                experimentalWatchApi: true
+                projectReferences: true
               }
             }
           ]
         }
       ]
     },
     plugins: [
-      new LiveReloadPlugin()
+      // See https://github.com/TypeStrong/ts-loader#usage-with-webpack-watch
+      new webpack.WatchIgnorePlugin({
+        paths: [
+          // Ignore generated .js, .map and .d.ts output files
+          /tinymce\/modules\/.*\/lib\/.*\.(js|map|d\.ts)$/,
+          // Something seems to trigger that node module package.json files change when they
+          // haven't, so lets just ignore them entirely
+          /node_modules\/.*\/package\.json$/
+        ]
+      }),
     ],
     output: {
-      filename: typeof entries === 'string' ? filename : "[name]/" + filename,
+      filename: '[name]',
       path: path.resolve(outDir),
       pathinfo: false
+    },
+    stats: {
+      assets: false,
+      modulesSpace: 5
     }
   };
 };
 
-let buildDemoEntries = (typeNames, type, demo) => typeNames.reduce(
+const buildDemoEntries = (typeNames, type, demo, pathPrefix = '') => typeNames.reduce(
   (acc, name) => {
-    var tsfile = `src/${type}/${name}/demo/ts/demo/${demo}`;
-    if (fs.existsSync(tsfile)) { acc[name] = tsfile; }
+    const tsfile = `src/${type}/${name}/demo/ts/demo/${demo}`;
+    if (fs.existsSync(tsfile)) { acc[`${pathPrefix}${type}/${name}/demo.js`] = tsfile; }
     return acc;
   }, {}
 );
 
-let buildEntries = (typeNames, type, entry) => typeNames.reduce(
+const buildEntries = (typeNames, type, entry, pathPrefix = '') => typeNames.reduce(
   (acc, name) => {
-    acc[name] = `src/${type}/${name}/main/ts/${entry}`;
+    const fileName = type.replace(/s$/, '') + '.js';
+    acc[`${pathPrefix}${type}/${name}/${fileName}`] = `src/${type}/${name}/main/ts/${entry}`;
     return acc;
   }, {}
 );
 
-let createPlugin = (name) => {
-  return create(`src/plugins/${name}/demo/ts/demo/Demo.ts`, 'tsconfig.plugin.json', `scratch/demos/plugins/${name}/`, 'demo.js');
+const createPlugin = (name) => {
+  return create({ 'demo.js': `src/plugins/${name}/demo/ts/demo/Demo.ts` }, '../../tsconfig.demo.json', `scratch/demos/plugins/${name}/`);
 };
 
-let createTheme = (name) => {
-  return create(`src/themes/${name}/demo/ts/demo/Demos.ts`, 'tsconfig.theme.json', `scratch/demos/themes/${name}`, 'demo.js');
+const createTheme = (name) => {
+  return create({ 'demo.js': `src/themes/${name}/demo/ts/demo/Demos.ts` }, '../../tsconfig.demo.json', `scratch/demos/themes/${name}`);
 };
 
-let createModel = (name) => {
-  return create(`src/models/${name}/demo/ts/demo/Demo.ts`, 'tsconfig.model.json', `scratch/demos/models/${name}`, 'demo.js');
+const createModel = (name) => {
```

Comment:
you mean the lack of `../..`? hmm. I didn't notice that. The demos all seem to work, but maybe they're working for some other reason.

Reply:
Yes it'll need fixing. The reason all the others use `../../` is because the `tsconfig.model.json` doesn't have the project references setup. So if you change something in any of the monorepo libraries it won't compile it.

Reply:
You know what? I don't think we actually use this, which is why I didn't notice. I might just delete the unused parts.